jgrapht-core/src/main/java/org/jgrapht/GraphTests.java

```diff
@@ -255,7 +255,7 @@
      * @param <E> the graph edge type
      */
     public static <V, E> boolean isBipartitePartition(
-        Graph<V, E> graph, Set<? extends V> firstPartition, Set<? extends V> secondPartition)
+        Graph<V, E> graph, Collection<? extends V> firstPartition, Collection<? extends V> secondPartition)
```

Comment:
No need for duplicates here. The method explicitly uses the fact that it accepts a set, by testing whether the cardinality of the vertex set of the graph equals the sum of the cardinalities of the two partitions. 

If we want to make this a `Collection`, additional checks need to be made inside the function. Otherwise, the callers should change instead of the callee.